src/app/IlcIntl.ts

```diff
@@ -0,0 +1,178 @@
+        return this.adapter.get();
+    }
+
+    public getDefault() {
+        return this.adapter.config.default;
+    }
+
+    public getSupported() {
+        return this.adapter.config.supported;
+    }
+
+    /**
+     * Allows to change locale or currency for the whole page
+     *
+     * @param config
+     */
+    public set(config: types.IntlConfig): Promise<void> {
+        if (!this.adapter.set) {
+            throw new Error("Looks like you're trying to call CSR only method during SSR.");
+        }
+
+        return this.adapter.set(config);
+    }
+
+    public localizeUrl(url: string, configOverride: types.IntlConfig = {}): string {
+        return IlcIntl.localizeUrl(this.adapter.config, url, { ...this.adapter.get(), ...configOverride });
+    }
+
+    /**
+     * Allows to parse URL and receive "non-localized" URL and information about locale that was encoded in URL.
+     *
+     * @param url
+     */
+    public parseUrl(url: string): { locale: string; cleanUrl: string } {
+        return IlcIntl.parseUrl(this.adapter.config, url);
+    }
+
+    /**
+     * [CSR ONLY] Allows to watch changes to locale or currency that are happening at the client side.
+     * @param callback
+     */
+    public watch(callback: (event: types.IntlUpdateEvent) => void): () => void {
+        if (!window.addEventListener) {
+            throw new Error("Looks like you're trying to call CSR only method during SSR.");
+        }
+
+        window.addEventListener(IlcIntl.eventName, callback as EventListener);
+        this.listeners.push(callback);
+
+        return () => {
+            for (const row of this.listeners) {
+                if (row === callback) {
+                    window.removeEventListener(IlcIntl.eventName, row);
+                    this.listeners.slice(this.listeners.indexOf(callback), 1);
+                    break;
+                }
+            }
+        };
+    }
+
+    /**
+     * [CSR ONLY] In apps that are running under ILC it shouldn't be used directly.
+     */
+    public unmount() {
+        if (!window.addEventListener) {
+            throw new Error("Looks like you're trying to call CSR only method during SSR.");
+        }
+
+        for (const callback of this.listeners) {
+            window.removeEventListener(IlcIntl.eventName, callback);
+        }
+
+        this.listeners = [];
+    }
+
+    static localizeUrl(config: types.IntlAdapterConfig, url: string, configOverride: types.IntlConfig = {}): string {
+        url = IlcIntl.parseUrl(config, url).cleanUrl;
+
+        const receivedLocale = configOverride.locale || config.default.locale;
+        const loc = IlcIntl.getCanonicalLocale(receivedLocale, config.supported.locale);
+        if (loc === null) {
+            throw new Error(`Unsupported locale passed. Received: "${receivedLocale}"`);
+        }
+
+        if (loc !== config.default.locale) {
+            url = `/${IlcIntl.getShortenedLocale(loc, config.supported.locale)}${url}`;
```

Comment:
I suggest using `url-join` here

Reply:
It will increase bundle size :( Don't see any big benefit here...